### PR TITLE
DAOS-9644 doc: correct provider documentation in server.yml (#8934)

### PR DESCRIPTION
**Pull request by Michael-Hennecke**
correct provider documentation in the daos_server.yml template

Doc-only: true
Signed-off-by: Michael Hennecke <michael.hennecke@intel.com>

### CODE REVIEW

File: utils/config/daos_server.yml [hunk 1]
```diff
@@ -70,12 +70,13 @@
...
 #
-## Force a specific libfabric provider to be used by all the engines.
-## The default provider depends on the interfaces that will be auto-detected:
-##  ofi+verbs for Infiniband/RoCE and
-##  ofi+tcp for non-RDMA-capable Ethernet.
+## Set the libfabric provider to be used by all the engines.
+## There is no default - run "daos_server network scan" to list the
+## providers that are supported on the fabric interfaces. Examples:
+##  ofi+verbs;ofi_rxm  for Infiniband/RoCE and
+##  ofi+tcp            for non-RDMA-capable fabrics
```
Comment (knard-intel) [inline utils/config/daos_server.yml:77]: From what I understand
```suggestion
##  ofi+tcp;ofi_rxm            for non-RDMA-capable fabrics
```

Comment (Michael-Hennecke) [reply utils/config/daos_server.yml:77]: that's not my understanding ... see "daos_server network scan -p all"

Comment (johannlombardi) [reply utils/config/daos_server.yml:77]: ofi_rxm is automatically added when ofi+tcp or ofi+verbs is specified as a provider. That's true for 2.2 and beyond.

**Review by johannlombardi — CHANGES_REQUESTED** [review]
must be consistent between verbs and tcp (add rxm for both or none).

File: utils/config/daos_server.yml (hunk 1 above)
Comment (Michael-Hennecke) [reply utils/config/daos_server.yml:77]: i see, the network scan output of 2.0 and 2.2 differs in this respect: v2.0 only reports `ofi+tcp`, whereas 2.2 reports both `ofi+tcp` and `ofi+tcp;ofi_rxm`.
will make this consistent by explicitly listing `;ofi_rxm` for verbs and tcp for all releases...

**Review by kccain — DISMISSED** [review]

**Review by knard-intel — DISMISSED** [review]
Fine for me.

**Review by johannlombardi — CHANGES_REQUESTED** [review]
I am afraid that if you re-add rxm to the "default value", then you are going to have to update src/control//server/config/server_test.go.

**Review by knard-intel — APPROVED** [review]
OK for me

**Review by johannlombardi — APPROVED** [review]

**Pull request merged**